Sources/TuistGenerator/Generator/InfoPlistContentProvider.swift

```diff
@@ -0,0 +1,151 @@
+    ///   - extendedWith: Values provided by the user to extend the default ones.
+    /// - Returns: Content to generate the Info.plist file.
+    func content(target: Target, extendedWith: [String: InfoPlist.Value]) -> [String: Any]? {
+        if target.product == .staticLibrary || target.product == .dynamicLibrary {
+            return nil
+        }
+
+        var content = base()
+
+        // Bundle package type
+        extend(&content, with: bundlePackageType(target))
+
+        // iOS app
+        if target.product == .app, target.platform == .iOS {
+            extend(&content, with: iosApp())
+        }
+
+        // macOS app
+        if target.product == .app, target.platform == .iOS {
+            extend(&content, with: macosApp())
+        }
+
+        // macOS
+        if target.platform == .macOS {
+            extend(&content, with: macos())
+        }
+
+        extend(&content, with: extendedWith.unwrappingValues())
+
+        return content
+    }
+
+    /// Returns a dictionary that contains the base content that all Info.plist
+    /// files should have regardless of the platform or product.
+    ///
+    /// - Returns: Base content.
+    func base() -> [String: Any] {
+        return [
+            "CFBundleDevelopmentRegion": "$(DEVELOPMENT_LANGUAGE)",
+            "CFBundleExecutable": "$(EXECUTABLE_NAME)",
+            "CFBundleIdentifier": "$(PRODUCT_BUNDLE_IDENTIFIER)",
+            "CFBundleInfoDictionaryVersion": "6.0",
+            "CFBundleName": "$(PRODUCT_NAME)",
+            "CFBundleShortVersionString": "1.0",
+            "CFBundleVersion": "1",
+        ]
+    }
+
+    /// Returns the Info.plist content that includes the CFBundlePackageType
+    /// attribute depending on the target product type.
+    ///
+    /// - Parameter target: Target whose Info.plist's CFBundlePackageType will be returned.
+    /// - Returns: Dictionary with the CFBundlePackageType attribute.
+    func bundlePackageType(_ target: Target) -> [String: Any] {
+        var packageType: String?
+
+        switch target.product {
+        case .app:
+            packageType = "APPL"
+        case .staticLibrary, .dynamicLibrary:
+            packageType = nil
+        case .uiTests, .unitTests, .bundle:
+            packageType = "BNDL"
+        case .staticFramework, .framework:
+            packageType = "FMWK"
+        }
+
+        if let packageType = packageType {
+            return ["CFBundlePackageType": packageType]
+        } else {
+            return [:]
+        }
+    }
+
+    /// Returns the default Info.plist content that iOS apps should have.
+    ///
+    /// - Returns: Info.plist content.
+    func iosApp() -> [String: Any] {
+        return [
+            "LSRequiresIPhoneOS": true,
+            "UILaunchStoryboardName": "LaunchScreen",
+            "UIMainStoryboardFile": "Main",
+            "UIRequiredDeviceCapabilities": [
+                "armv7",
+            ],
+            "UISupportedInterfaceOrientations": [
```

Comment:
Do you think I should leave those out @tuist/core?

Reply:
Those are the default from Xcode so should be ok, what were your concerns?

The good news is users can override this by specifying the key in the manifest.

e.g.

```swift
 infoPlist: .extendingDefault(with: ["UISupportedInterfaceOrientations": [
  "UIInterfaceOrientationPortrait"
]]),
```

The current logic ensures the user defined values take precedence 👍